volmdlr/faces.py

```diff
@@ -2710,19 +2720,200 @@ def point2d_with_dimension_to_3d(self, point2d, points_x, points_y, xmin, xmax,
+                                                                     bsplinecurve2d.knot_multiplicities,
+                                                                     bsplinecurve2d.knots,
+                                                                     bsplinecurve2d.weights,
+                                                                     bsplinecurve2d.periodic)
+
+        return bsplinecurve2d_with_dimension
+
+
+    def bsplinecurve3d_to_2d_with_dimension(self, bsplinecurve3d, points_x, points_y):
+        '''
+        compute the bsplinecurve2d of a bsplinecurve3d, on a Bspline surface, in the dimensioned frame
+        '''
+
+        bsplinecurve2d_01 = self.bsplinecurve3d_to_2d(bsplinecurve3d)
+        bsplinecurve2d_with_dimension = self.bsplinecurve2d_parametric_to_dimension(bsplinecurve2d_01, points_x, points_y)
+
+        return bsplinecurve2d_with_dimension
+
+
+    def bsplinecurve2d_with_dimension_to_parametric_frame(self, bsplinecurve2d):
+        '''
+        convert a bsplinecurve2d from the dimensioned to the parametric frame
+        '''
+
+        points_dim = bsplinecurve2d.control_points
+        points = []
+        for key in self._grids2d:
+            [points_x, points_y, xmin, xmax, ymin, ymax] = key
+
+        for p in points_dim:
+            points.append(self.point2d_with_dimension_to_parametric_frame(p, points_x, points_y, xmin, xmax, ymin, ymax))
+
+        bsplinecurve2d = volmdlr.edges.BSplineCurve2D(bsplinecurve2d.degree, points,
+                                                      bsplinecurve2d.knot_multiplicities,
+                                                      bsplinecurve2d.knots,
+                                                      bsplinecurve2d.weights,
+                                                      bsplinecurve2d.periodic)
+        return bsplinecurve2d
+
+
+    def bsplinecurve2d_with_dimension_to_3d(self, bsplinecurve2d):
+        '''
+        compute the bsplinecurve3d, on a Bspline surface, of a bsplinecurve2d defined in the dimensioned frame
+        '''
+
+        bsplinecurve2d_01 = self.bsplinecurve2d_with_dimension_to_parametric_frame(bsplinecurve2d)
+        bsplinecurve3d = self.bsplinecurve2d_to_3d(bsplinecurve2d_01)
+
+        return bsplinecurve3d
+
+
+    def arc2d_parametric_to_dimension(self, arc2d, points_x, points_y):
+        '''
+        convert a arc2d from the parametric to the dimensioned frame
+        '''
+
+        xmin, xmax, ymin, ymax = 0, 1, 0, 1
+
+        start = self.point2d_parametric_to_dimension(arc2d.start, points_x, points_y, xmin, xmax, ymin, ymax)
+        end = self.point2d_parametric_to_dimension(arc2d.end, points_x, points_y, xmin, xmax, ymin, ymax)
+        interior = self.point2d_parametric_to_dimension(arc2d.interior, points_x, points_y, xmin, xmax, ymin, ymax)
+
+        arc2d_with_dimension = volmdlr.edges.Arc2D(start, interior, end)
+
+        return arc2d_with_dimension
```

Comment:
Are you sure an Arc2D in parametric coordinates is also an Arc2D in dimensional coordinates ?